openwisp_users/admin.py

```diff
@@ -41,6 +42,24 @@ class OrganizationUserInline(admin.StackedInline):
 class EmailRequiredMixin(forms.ModelForm):
     email = forms.EmailField(label=_('Email'), max_length=254, required=True)
 
+    def _clean_email(self, email):
+        try:
+            EmailAddress.objects.get(email=email)
+            raise ValidationError({'email': ['User with this email already exists.']})
+        except EmailAddress.DoesNotExist:
+            pass
+
+    def clean(self):
+        cleaned_data = super(EmailRequiredMixin, self).clean()
+        # check if editing
+        try:
+            EmailAddress.objects.get(email=cleaned_data.get('email'))
+            if 'email' in self.changed_data:
+                self._clean_email(email=cleaned_data.get('email'))
+        except EmailAddress.DoesNotExist:
+            self._clean_email(email=cleaned_data.get('email'))
```

Comment:
why are you performing identical operations to the ones performed in `_clean_email`?
Can't you just do:

```python
if 'email' in self.changed_data:
    self._clean_email(email=cleaned_data.get('email'))
```